src/main/java/org/purescript/file/PSFile.kt

```diff
@@ -16,10 +16,15 @@ class PSFile(viewProvider: FileViewProvider) :
         return "Purescript File"
     }
 
-    val module: PSModule
-        get() = findChildByClass(PSModule::class.java)!!
+    /**
+     * @return the [PSModule] that this file contains,
+     * or null if the module couldn't be parsed
+     */
+    val module: PSModule?
+        get() = findChildByClass(PSModule::class.java)
```

Comment:
This is the change that I wanted to make. The rest is either handling of possible null modules, and some removal of boilerplate in the tests.